jujugui/static/gui/src/app/components/search-results/search-results.js

```diff
@@ -228,7 +228,11 @@ YUI.add('search-results', function(Y) {
       }
       this._changeActiveComponent('loading');
       this.setState({ waitingForSearch: true });
-
+      // If there is an exist search request then abort it before starting the
+      // new search.
+      if (this.searchXhr) {
```

Comment:
This is a good addition to have, but maybe some more investigation is necessary to find out why a pre-existing search is happening on the initial load, and that should be fixed.